src/arrayMethodSort.js

```diff
@@ -4,8 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const arr = this;
+
+    const sortFunction = (a, b) => {
+      const strA = String(a);
+      const strB = String(b);
+
+      if (strA < strB) {
+        return -1;
+      }
+
+      if (strA > strB) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    for (let i = 0; i < arr.length; i++) {
+      for (let j = 0; j < arr.length - 1 - i; j++) {
+        if (compareFunction === undefined) {
+          if (sortFunction(arr[j], arr[j + 1]) > 0) {
+            [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
+          }
+        } else {
+          if (compareFunction(arr[j], arr[j + 1]) > 0) {
+            [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
+          }
+        }
+      }
+    }
```

Comment:
The implementation uses a bubble sort algorithm, which is simple and works correctly for this task. However, bubble sort is not efficient for large arrays. Consider using more efficient algorithms like quicksort or mergesort if performance becomes an issue.